### ISSUE TITLE
x/distribution withdraw-all-rewards got wrong account sequence on PeriodicVestingAccount

### ISSUE THREAD

**Issue by kwunyeung** [desmos-labs/desmos#567]
The buggy account is

https://morpheus.desmos.network/accounts/desmos1snmetx9sfrmznr4z6d3lnwlyvwu36sgcpzzhlr

I ran 

```
desmos tx distribution withdraw-all-rewards
```
to withdraw all rewards from all delegations but then after broadcasting the tx, it gave me this

```
Error: rpc error: code = InvalidArgument desc = account sequence mismatch, expected 29, got 28: incorrect account sequence: invalid request
```

The sequence number is incremented before the tx is committed. Interestingly, for the same account, withdrawing rewards from each delegation by `MsgWithdrawDelegatorReward` has no problem. It only happens with the `withdraw-all-rewards` command which broadcast a tx with multiple `MsgWithdrawDelegatorReward`. 

I have tested this with other accounts but only this account would give the issue. It has 12 delegations and it's a `PeriodicVestingAccount`. Not sure if this is an issue in `x/distribution` in the SDK or some misuse of `x/auth`.

**Comment by dadamu** [comment]
@kwunyeung The `withdraw-all-rewards` cmd should have been fixed in [this PR](https://github.com/cosmos/cosmos-sdk/issues/9489) with 0.42.5 version of cosmos-sdk. By the way, I can't reproduce it as well in current desmos version.

**Comment by kwunyeung** [comment]
@dadamu the issue exists when an account has more than 5 delegations as `withdraw-all-rewards` will commit the first tx with 5 `MsgWithdrawDelegatorReward ` messages and then another 5 `MsgWithdrawDelegatorReward` messages, etc. The sequence will not be correct if the `--broadcast-mode` is not `block`. 

If this has been fixed in `v0.42.5`, have we picked the fix in our sdk fork? I think `withdraw-all-rewards` should bypass `sync` and `async` mode.

**Comment by dadamu** [comment]
@kwunyeung Tested it with 12 nodes. The default max-msgs is 5 so the error happens, setting it to 0 (meaning that send all msgs in one transaction) can solve it.

**Comment by kwunyeung** [comment]
I guess we need to mention it somewhere in the doc then.

**Comment by RiccardoM** [comment]
I've opened a PR to improve the UX of the `withdraw-all-rewards` directly inside the Cosmos SDK: https://github.com/cosmos/cosmos-sdk/pull/9781

**Comment by RiccardoM** [comment]
Should have been solved with `v0.17.6`. In that version I'm using commit desmos-labs/cosmos-sdk@640390b3cc37f50f84a56c20f063905137fefffc our fork of the SDK repo, which also integrates the commit that should have fixed this (desmos-labs/cosmos-sdk@52843ecb8bb2dd8abe1366bb0b11cd70cd1fcc82)
